File: SEM_4/CSW Sem-4/Assignment 5/Q9.java
import java.util.Scanner;
public class Q9 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        try {
            System.out.print("Enter the dividend: ");
            int dividend = sc.nextInt();

            System.out.print("Enter the divisor: ");
            int divisor = sc.nextInt();

            double result = divide(dividend, divisor);
            System.out.println("Result of division: " + result);
        } catch (ArithmeticException e) {
            System.out.println("ArithmeticException occurred: " + e.getMessage());
        } catch (Exception e) {
            System.out.println("An error occurred: " + e.getMessage());
        } finally {
            sc.close();
        }
    }
    
    public static double divide(int dividend, int divisor) {
        if (divisor == 0) {
            throw new ArithmeticException("Division by zero is not allowed.");
        }

        return (double) dividend / divisor;
    }
}

File: SEM_5/Python/Assignment-8/Q9.py
def safe_divide(num1, num2):
    try:
        num1 = float(num1)
        num2 = float(num2)
        if num2 == 0:
            raise ValueError("Error: Division by zero is not allowed.")
        result = num1 / num2
        return result
    except ValueError as ve:
        return str(ve)
    except Exception as e:
        return f"Error: {e}"
print(safe_divide(10, 2))
print(safe_divide(10, 0))
print(safe_divide('a', 2))
print(safe_divide(10, 'b'))
print(safe_divide('a', 'b'))